app/src/main/java/id/rizmaulana/covid19/ui/percountry/indonesia/CountryIndonesiaActivity.kt

```diff
@@ -0,0 +1,86 @@
+package id.rizmaulana.covid19.ui.percountry.indonesia
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.view.View
+import id.rizmaulana.covid19.databinding.ActivityCountryIndonesiaBinding
+import id.rizmaulana.covid19.ui.adapter.ItemTypeFactoryImpl
+import id.rizmaulana.covid19.ui.adapter.VisitableRecyclerAdapter
+import id.rizmaulana.covid19.ui.adapter.viewholders.DailyItem
+import id.rizmaulana.covid19.ui.adapter.viewholders.TextItem
+import id.rizmaulana.covid19.ui.base.BaseActivity
+import id.rizmaulana.covid19.ui.base.BaseViewItem
+import id.rizmaulana.covid19.ui.dailygraph.DailyGraphActivity
+import id.rizmaulana.covid19.util.ext.observe
+import kotlinx.android.synthetic.main.activity_dashboard.*
+import org.koin.android.viewmodel.ext.android.viewModel
+
+class CountryIndonesiaActivity : BaseActivity() {
+    private val viewModel by viewModel<CountryIndonesiaViewModel>()
+    private lateinit var binding: ActivityCountryIndonesiaBinding
+    private val viewAdapter by lazy {
+        VisitableRecyclerAdapter(
+            ItemTypeFactoryImpl(),
+            ::onItemClicked
+        )
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityCountryIndonesiaBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        setupActionBarWithBackButton(binding.toolbar)
+        initView()
+
+        viewModel.loadData()
+
+    }
+
+    private fun initView() {
+        with(binding.recyclerView) {
+            adapter = viewAdapter
+            setHasFixedSize(true)
+        }
+        swipe_refresh.setOnRefreshListener {
```

Comment:
use the binding to access `swipe_refresh` @rizmaulana, remove kotlin synthetic usage